File: scripts/Royal/npc/Mailbox.js
/*
Xenia Mailbox
*/
importPackage(Packages.objects.item);
importPackage(Packages.objects.users);

importPackage(Packages.database);

purple = "#fMap/MapHelper.img/weather/starPlanet/7#";
blue = "#fMap/MapHelper.img/weather/starPlanet/8#";
starBlue = "#fUI/GuildMark.img/Mark/Pattern/00004001/11#"
starYellow = "#fUI/GuildMark.img/Mark/Pattern/00004001/3#"
starWhite = "#fUI/GuildMark.img/Mark/Pattern/00004001/15#"
starBrown = "#fUI/GuildMark.img/Mark/Pattern/00004001/5#"
starRed = "#fUI/GuildMark.img/Mark/Pattern/00004001/1#"
starBlack = "#fUI/GuildMark.img/Mark/Pattern/00004001/16#"
starPurple = "#fUI/GuildMark.img/Mark/Pattern/00004001/13#"
star = "#fUI/FarmUI.img/objectStatus/star/whole#"
S = "#fUI/CashShop.img/CSEffect/today/0#"
reward = "#fUI/UIWindow2.img/Quest/quest_info/summary_icon/reward#"
obtain = "#fUI/UIWindow2.img/QuestIcon/4/0#"
color = "#fc0xFF6600CC#"
black = "#fc0xFF000000#"
enter = "\r\n"
enter2 = "\r\n\r\n"

var status = -1;

function start() {
    status = -1;
    action(1, 0, 0);
}

function action(mode, type, selection) {
    if (mode == 1) {
        status--;
    } else {
        if (status == 0) {
            cm.dispose();
        }
        status--;
    }
    if (status == 0) {
        say = "#fs11#" + star + color + "คุณต้องการรับไอเท็มจากกล่องจดหมายหรือไม่?" + enter2 + starRed + "#r กรุณาทำช่องว่างในช่องเก็บของให้เพียงพอ\r\n\r\n#e< รายการในกล่องจดหมาย >#n\r\n"
        try {
            con = DBConnection.getConnection();
            ps = con.prepareStatement("SELECT * FROM `offline` WHERE `chrid` = " + cm.getPlayer().getId() + " AND status = 0");
            rs = ps.executeQuery();
            while (rs.next()) {
                say += black + "#i" + rs.getString("item") + "#" + color + " #z" + rs.getString("item") + "# " + rs.getInt("qua") + " ชิ้น\r\n";
            }
            rs.close();
            ps.close();
            con.close();
            cm.sendYesNo(say);
            return;
        } catch (e) {
            cm.sendOk("เกิดข้อผิดพลาด\r\n\r\n" + e);
            cm.dispose();
            return;
        }
    } else if (status == 1) {
        if (cm.getPlayer().getInventory(Packages.objects.item.MapleInventoryType.EQUIP).getNumFreeSlot() < 10 || cm.getPlayer().getInventory(Packages.objects.item.MapleInventoryType.USE).getNumFreeSlot() < 10 || cm.getPlayer().getInventory(Packages.objects.item.MapleInventoryType.ETC).getNumFreeSlot() < 10) {
            cm.sendOk("#fs11#กรุณาทำช่องว่างในช่อง Equip, Use, และ Etc อย่างน้อย 10 ช่อง");
            cm.dispose();
            return;
        }

        if (MailBox.pickUpItemOff(cm.getPlayer())) {
            MailBox.removeItemOff(cm.getPlayer().getId());
            cm.sendOk("#fs11#คุณได้รับไอเท็มจากกล่องจดหมายเรียบร้อยแล้ว\r\nกรุณาตรวจสอบช่องเก็บของ");
        } else {
            cm.sendOk("#fs11#ไม่มีไอเท็มในกล่องจดหมาย");
        }
        cm.dispose();
    }
}

File: src/objects/users/MailBox.java
package objects.users;

import database.DBConnection;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class MailBox {
   public static int getCid(String name) {
      int ret = 0;

      try (Connection con = DBConnection.getConnection()) {
         PreparedStatement ps = con.prepareStatement("SELECT * FROM characters WHERE name = ?");
         ps.setString(1, name);
         ResultSet rs = ps.executeQuery();

         while (rs.next()) {
            ret = rs.getInt("id");
         }

         ps.close();
         rs.close();
      } catch (SQLException var7) {
         var7.printStackTrace();
      }

      return ret;
   }

   public static void removeItemOff(int chr) {
      try (Connection con = DBConnection.getConnection()) {
         PreparedStatement ps = con.prepareStatement("SELECT * FROM offline WHERE chrid = ?");
         ps.setInt(1, chr);
         ResultSet rs = ps.executeQuery();

         while (rs.next()) {
            ps = con.prepareStatement("UPDATE offline SET `status` = 1 WHERE chrid = ?");
            ps.setInt(1, chr);
            ps.execute();
         }

         ps.close();
         rs.close();
      } catch (SQLException var6) {
         var6.printStackTrace();
      }
   }

   public static boolean pickUpItemOff(MapleCharacter chr) {
      try {
         boolean var4;
         try (Connection con = DBConnection.getConnection()) {
            PreparedStatement ps = con.prepareStatement("SELECT * FROM offline WHERE chrid = ? AND `status` = 0");
            ps.setInt(1, chr.getId());
            ResultSet rs = ps.executeQuery();
            if (rs.first()) {
               do {
                  chr.gainItem(rs.getInt("item"), rs.getInt("qua"));
               } while (rs.next());

               ps.close();
               rs.close();
               return true;
            }

            ps.close();
            rs.close();
            con.close();
            var4 = false;
         }

         return var4;
      } catch (SQLException var7) {
         var7.printStackTrace();
         return true;
      }
   }

   public static boolean isItemOff(MapleCharacter chr) {
      try {
         boolean var4;
         try (Connection con = DBConnection.getConnection()) {
            PreparedStatement ps = con.prepareStatement("SELECT * FROM offline WHERE chrid = ? AND `status` = 0");
            ps.setInt(1, chr.getId());
            ResultSet rs = ps.executeQuery();
            if (rs.first()) {
               ps.close();
               rs.close();
               return true;
            }

            ps.close();
            rs.close();
            con.close();
            var4 = false;
         }

         return var4;
      } catch (SQLException var7) {
         var7.printStackTrace();
         return true;
      }
   }

   public static void sendItemOff(int chr, int item, int qua) {
      try (Connection con = DBConnection.getConnection()) {
         PreparedStatement pse = con.prepareStatement("INSERT INTO offline (chrid, item, qua) VALUES (?, ?, ?)");
         pse.setInt(1, chr);
         pse.setInt(2, item);
         pse.setInt(3, qua);
         pse.executeUpdate();
         pse.close();
      } catch (SQLException var8) {
         var8.printStackTrace();
      }
   }
}
